File: analogagenda.client/src/app/DTOs/user-settings.dto.ts
export interface UserSettingsDto {
  userId: string;
  isSubscribed: boolean;
  currentFilmId?: string | null;
  tableView: boolean;
  entitiesPerPage: number;
  /** Preferred home dashboard section order; omit on PATCH when not changing layout. */
  homeSectionOrder?: string[] | null;
}

File: Database/DTOs/UserSettingsDto.cs
namespace Database.DTOs;

public class UserSettingsDto
{
    public string UserId { get; set; } = string.Empty;

    public bool IsSubscribed { get; set; }

    public string? CurrentFilmId { get; set; }

    public bool TableView { get; set; }

    public int EntitiesPerPage { get; set; }

    /// <summary>Preferred order of home dashboard sections. Null = server default / not sent on PATCH.</summary>
    public string[]? HomeSectionOrder { get; set; }
}

